tests/test_shells.py

```diff
@@ -0,0 +1,53 @@
+import json
+from urllib import urlencode
+from mock import patch
+
+import pytest
+from tornado.httpclient import HTTPError
+
+from fixtures import standard_graph, graph, users, groups, session, permissions  # noqa
+from fixtures import fe_app as app  # noqa
+from grouper import public_key
+from grouper.model_soup import User, Request, AsyncNotification
+from grouper.constants import USER_METADATA_SHELL_KEY
+from url_util import url
+
+
+@pytest.mark.gen_test
+def test_shell(session, users, http_client, base_url):
+    user = users['zorkian@a.co']
+    assert not user.get_metadata(USER_METADATA_SHELL_KEY)
+
+    user = User.get(session, name=user.username)
+    fe_url = url(base_url, '/users/{}/shell'.format(user.username))
+    resp = yield http_client.fetch(fe_url, method="POST",
+            body=urlencode({'shell': "/bin/bash"}),
+            headers={'X-Grouper-User': user.username})
+    assert resp.code == 200
+
+    user = User.get(session, name=user.username)
+
+    assert user.get_metadata(USER_METADATA_SHELL_KEY) is not None, "The user should have shell metadata"
```

Comment:
also i noticed these lines exceeded 100 characters. guess we should `flake8` the test directory as well. :)

mind wrapping these?